.ci/test.sh

```diff
@@ -220,14 +226,14 @@ if [[ $TASK == "gpu" ]]; then
         pip install \
             -v \
             --config-settings=cmake.define.USE_GPU=ON \
-            ./dist/lightgbm-$LGB_VER.tar.gz \
+            "./dist/lightgbm-$LGB_VER.tar.gz" \
         || exit 1
         pytest ./tests/python_package_test || exit 1
         exit 0
     elif [[ $METHOD == "wheel" ]]; then
         sh ./build-python.sh bdist_wheel --gpu || exit 1
         sh ./.ci/check-python-dists.sh ./dist || exit 1
-        pip install ./dist/lightgbm-$LGB_VER*.whl -v || exit 1
+        pip install ./dist/lightgbm-"$LGB_VER"*.whl -v || exit 1
```

Comment:
```suggestion
        pip install "$(echo ./dist/lightgbm-${LGB_VER}*.whl)" -v || exit 1
```

Please, as I've said in other reviews, do not add `""` in the middle of paths.

Reply:
Gotcha. Note that L236 261 and 281 are now all raising a https://www.shellcheck.net/wiki/SC2086 warning

Reply:
Please, just use another set of quotes similar to what you did in #6621

https://github.com/microsoft/LightGBM/blob/41ba9e8f00c89d72e5cb71c964722ce1ed4d8445/.ci/check-python-dists.sh#L23

> Can we use a find + exec here?

No, I don't think that complexity is justified here.